build_tools/cmake/iree_macros.cmake

```diff
@@ -359,3 +359,52 @@ function(iree_validate_required_arguments
     endif()
   endforeach()
 endfunction()
+
+# iree_compile_flags_for_patform
+#
+# Helper function to add necessary compile flags based on platform-specific
+# configurations. Note the flags are added for cpu backends only.
+function(iree_compile_flags_for_platform OUT_FLAGS IN_FLAGS)
+  if(NOT (IN_FLAGS MATCHES "iree-hal-target-backends=llvm-cpu" OR
+          IN_FLAGS MATCHES "iree-hal-target-backends=vmvx"))
+    set(${OUT_FLAGS} "" PARENT_SCOPE)
+    return()
+  endif()
+
+  if(ANDROID AND NOT IN_FLAGS MATCHES "iree-llvm-target-triple")
+    # Android's CMake toolchain defines some variables that we can use to infer
+    # the appropriate target triple from the configured settings:
+    # https://developer.android.com/ndk/guides/cmake#android_platform
+    #
+    # In typical CMake fashion, the various strings are pretty fuzzy and can
+    # have multiple values like "latest", "android-25"/"25"/"android-N-MR1".
+    #
+    # From looking at the toolchain file, ANDROID_PLATFORM_LEVEL seems like it
+    # should pretty consistently be just a number we can use for target triple.
+    set(_TARGET_TRIPLE "aarch64-none-linux-android${ANDROID_PLATFORM_LEVEL}")
+    list(APPEND _FLAGS "--iree-llvm-target-triple=${_TARGET_TRIPLE}")
+  endif()
+
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+     CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+     NOT IN_FLAGS MATCHES "iree-llvm-target-triple")
+    # RV64 Linux crosscompile toolchain can support iree-compile with
+    # specific CPU flags. Add the llvm flags to support RV64 RVV codegen if
+    # llvm-target-triple is not specified.
+    list(APPEND _FLAGS ${RISCV64_TEST_DEFAULT_LLVM_FLAGS})
+  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv32" AND
+         CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+         NOT IN_FLAGS MATCHES "iree-llvm-target-triple")
+    # RV32 Linux crosscompile toolchain can support iree-compile with
+    # specific CPU flags. Add the llvm flags to support RV32 RVV codegen if
+    # llvm-target-triple is not specified.
+    list(APPEND _FLAGS ${RISCV32_TEST_DEFAULT_LLVM_FLAGS})
+  endif()
+
+  if(EMSCRIPTEN AND NOT IN_FLAGS MATCHES "iree-llvm-target-triple")
+    list(APPEND _FLAGS "--iree-llvm-target-triple=wasm32-unknown-emscripten")
+    list(APPEND _FLAGS "--iree-llvm-target-cpu-features=+atomics,+bulk-memory")
+  endif()
```

Comment:
I'm not sure if `+atomics` and `+bulk-memory` are needed in general. We could leave this empty, or just set `+simd128` as I'd been using in the mnist static web sample: https://github.com/iree-org/iree/blob/906995e380ef2a868f03c3b5d06ce25bedf2209e/experimental/web/sample_static/build_sample.sh#L51-L59

For the dynamic sample I'd been using these: https://github.com/iree-org/iree/blob/906995e380ef2a868f03c3b5d06ce25bedf2209e/experimental/web/sample_dynamic/build_sample.sh#L48-L56

So generally, I think that until the web stuff has stabilized a bit, maybe just leave the features empty, unless strictly necessary? It probably doesn't matter much for tests. We could also follow what's been done for RISCV, with some "default llvm flags".

Reply:
Done.